network/network-address/src/encrypted.rs

```diff
@@ -0,0 +1,482 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::RawNetworkAddress;
+use aes_gcm::{
+    aead::{generic_array::GenericArray, AeadInPlace, NewAead},
+    Aes256Gcm,
+};
+use libra_crypto::{
+    compat::Sha3_256,
+    hash::HashValue,
+    hkdf::{Hkdf, HkdfError},
+};
+use move_core_types::account_address::AccountAddress;
+use once_cell::sync::Lazy;
+#[cfg(any(test, feature = "fuzzing"))]
+use proptest::prelude::*;
+use serde::{Deserialize, Serialize};
+use std::{convert::TryFrom, mem};
+use thiserror::Error;
+
+/// The length in bytes of the AES-256-GCM authentication tag.
+pub const AES_GCM_TAG_LEN: usize = 16;
```

Comment:
btw I also set this in crypto/noise IIRC, maybe worth setting this in one place in crypto

Reply:
yeah i remember that, just thought it was clearer to have it nearby the actual code that uses it